File: container-with-most-water/ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0,right = height.length - 1;
        int maxAmount = 0;
        
        while(left < right){
            int amount = calculateAmount(left,right,height);
            if(amount > maxAmount){
                maxAmount = amount;
            }

            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }
        
        return maxAmount;
    }

    private int calculateAmount(int left, int right, int [] height){
        return Math.min(height[left], height[right])*(right-left);
    }
}

File: container-with-most-water/ContainerWithMostWater.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var left:Int = 0;
        var right:Int = height.size-1;
        var maxAmount:Int = 0;

        while(left < right){
            var amount  = calculateAmount(left, right, height)

            if(amount > maxAmount) maxAmount = amount
            if(height[left] < height[right]) left++ else right--
        }
        return maxAmount
    }
    private fun calculateAmount(left:Int, right: Int, height:IntArray): Int
    {
        return kotlin.math.min(height[left],height[right]) * (right - left);
    }
}